packages/x-tree-view/src/internals/plugins/useTreeViewKeyboardNavigation/useTreeViewKeyboardNavigation.test.tsx

```diff
@@ -3,206 +3,517 @@ import { expect } from 'chai';
+        expect(response.getFocusedItemId()).to.equal('1');
+      });
+    });
+
+    describe('key: Home', () => {
+      it('should move the focus to the first item in the tree', () => {
+        const response = render({
+          items: [{ id: '1' }, { id: '2' }, { id: '3' }, { id: '4' }],
+        });
 
-        fireEvent.keyDown(getItemRoot('2'), { key: '4' });
-        expect(getFocusedItemId()).to.equal('4');
+        act(() => {
+          response.getItemRoot('4').focus();
+        });
+        fireEvent.keyDown(response.getItemRoot('4'), { key: 'Home' });
+        expect(response.getFocusedItemId()).to.equal('1');
       });
+    });
 
-      it('should not move focus when pressing a modifier key and a letter', () => {
-        const { getFocusedItemId, getItemRoot } = render({
+    describe('key: End', () => {
+      it('should live the focus to the last item in the tree when the last item is not expanded', () => {
+        const response = render({
           items: [
-            { id: '1', label: 'one' },
-            { id: '2', label: 'two' },
-            { id: '3', label: 'three' },
-            { id: '4', label: 'four' },
+            { id: '1' },
+            { id: '2' },
+            { id: '3' },
+            { id: '4', children: [{ id: '4.1' }, { id: '4.2' }] },
           ],
         });
 
         act(() => {
-          getItemRoot('1').focus();
+          response.getItemRoot('1').focus();
         });
-        expect(getFocusedItemId()).to.equal('1');
-
-        fireEvent.keyDown(getItemRoot('1'), { key: 't', ctrlKey: true });
-        expect(getFocusedItemId()).to.equal('1');
+        fireEvent.keyDown(response.getItemRoot('1'), { key: 'End' });
+        expect(response.getFocusedItemId()).to.equal('4');
+      });
 
-        fireEvent.keyDown(getItemRoot('1'), { key: 't', metaKey: true });
-        expect(getFocusedItemId()).to.equal('1');
+      it('should live the focus to the last item in the tree when the last item is expanded', () => {
+        const response = render({
+          items: [
+            { id: '1' },
+            { id: '2' },
+            { id: '3' },
+            { id: '4', children: [{ id: '4.1', children: [{ id: '4.1.1' }] }] },
+          ],
+          defaultExpandedItems: ['4', '4.1'],
+        });
 
-        fireEvent.keyDown(getItemRoot('1'), { key: 't', shiftKey: true });
-        expect(getFocusedItemId()).to.equal('1');
+        act(() => {
+          response.getItemRoot('1').focus();
+        });
+        fireEvent.keyDown(response.getItemRoot('1'), { key: 'End' });
+        expect(response.getFocusedItemId()).to.equal('4.1.1');
       });
+    });
 
-      it('should work on disabled item when disabledItemsFocusable={true}', () => {
-        const { getFocusedItemId, getItemRoot } = render({
+    describe('key: * (asterisk)', () => {
+      it('should expand all items that are at the same level as the current item', () => {
```

Comment:
Nitpick: having an extra test doing the same on a nested item could make much sense. 👍

Reply:
Clearly!
Added